streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/Joined.scala

```diff
@@ -39,4 +39,23 @@ object Joined {
                        otherValueSerde: Serde[VO]): JoinedJ[K, V, VO] =
     JoinedJ.`with`(keySerde, valueSerde, otherValueSerde)
 
+  /**
+   * Create an instance of [[org.apache.kafka.streams.kstream.Joined]] with key, value, and otherValue [[Serde]]
+   * instances.
+   * `null` values are accepted and will be replaced by the default serdes as defined in config.
+   *
+   * @tparam K              key type
+   * @tparam V              value type
+   * @tparam VO             other value type
+   * @param name            name of possible repartition topic
+   * @param keySerde        the key serde to use.
+   * @param valueSerde      the value serde to use.
+   * @param otherValueSerde the otherValue serde to use. If `null` the default value serde from config will be used
+   * @return new [[org.apache.kafka.streams.kstream.Joined]] instance with the provided serdes
+   */
+  def `with`[K, V, VO](
+    name: String
+  )(implicit keySerde: Serde[K], valueSerde: Serde[V], otherValueSerde: Serde[VO]): JoinedJ[K, V, VO] =
```

Comment:
I agree this is pretty ugly. The problem is that the whole line is too long.

I recommend putting this at the top of the file:
```
// disable spotless scala, which wants to make a mess of the argument lists
// format: off
```

This will disable both spotlessScalaCheck and spotlessApply, then you can do what you want with the arg list.

As far as actual formatting goes, I think you have three equally good options to choose from:

1. Just let it be long... it's one line, and I think the readability gained from having both arg lists on the same line outweighs the length of the line in this case:
```
  def `with`[K, V, VO](name: String)(implicit keySerde: Serde[K], valueSerde: Serde[V], otherValueSerde: Serde[VO]): JoinedJ[K, V, VO] =
```

2. put each arg list on its own line. It's still over 120 characters, but again, this is pretty readable:
```
  def `with`[K, V, VO](name: String)
                      (implicit keySerde: Serde[K], valueSerde: Serde[V], otherValueSerde: Serde[VO]): JoinedJ[K, V, VO] =
```

3. put every argument on its own line. This is now short enough... Is it readable enough? I leave to you:
```
  def `with`[K, V, VO](name: String)
                      (implicit keySerde: Serde[K], 
                       valueSerde: Serde[V], 
                       otherValueSerde: Serde[VO]): JoinedJ[K, V, VO] =
```